src/game.html

```diff
@@ -10054,18 +10064,36 @@
                 placeWelcomeBanner();
             }
 
-            merchant.isActiveOnFloor =
-                merchant.isAlive &&
-                (floor === 1 || Math.random() < 0.35);
+            if (merchant.isAlive) {
+                if (floor === 1) {
+                    // Always spawn on floor 1
+                    merchant.isActiveOnFloor = true;
+                    merchant.consecutiveFloorsAbsent = 0;
+                } else if (merchant.consecutiveFloorsAbsent >= 3) { // After 3 consecutive floors of an absent Merchant, force spawn
+                    
+                    merchant.isActiveOnFloor = true;
+                    merchant.consecutiveFloorsAbsent = 0;
+                } else {
+                    merchant.isActiveOnFloor = Math.random() < 0.35; // Chances at spawning normally 
+                    
+                    if (merchant.isActiveOnFloor) {
+                        merchant.consecutiveFloorsAbsent = 0;
+                    } else {
+                        merchant.consecutiveFloorsAbsent++;
+                    }
+                }
```

Comment:
This same logic could be handled in half as many lines:
```suggestion
                // Always spawn on the first floor, or after 3 floors of absence
                if (floor === 1 || merchant.consecutiveFloorsAbsent >= 3) {
                    merchant.isActiveOnFloor = true;
                    merchant.consecutiveFloorsAbsent = 0;
                } else {
                    merchant.isActiveOnFloor = Math.random() < 0.35;
                    merchant.consecutiveFloorsAbsent +=
                        merchant.isActiveOnFloor;
                }
```